src/dmd/declaration.d

```diff
@@ -733,6 +732,17 @@ extern (C++) final class AliasDeclaration : Declaration
         return new AliasDeclaration(loc, id, type);
     }
 
+    /** Returns: an AliasDeclaration with its set of storage classes adjusted so
+    that it is possible to detect that it is created to make a template parameter
+    visible in a template body.*/
+    static extern (D) AliasDeclaration createForTemplateParameter(T)(const ref Loc loc, Identifier ident, T t)
+    if (is(T == Type) || is(T == Dsymbol))
+    {
+        AliasDeclaration result = new AliasDeclaration(loc, ident, t);
+        result.storage_class |= STC.templateparameter;
+        return result;
+    }
```

Comment:
Make it an overload set

Reply:
yeah so the idea was 

1. dont use an optional trailing param, as discouraged by the contribution guidlines
2. adding a non-optional param is kinda strange as we only need to create the overload taking a trailing `bool` param with `true` as arg.

Reply:
does my argument change you mind ?

Reply:
I'm probably missing something here.
You have a static function that is templated, but the template argument can only be `Type` or `Dsymbol`, and those are disjoint types. What I suggested was, instead of having it a template, to make two functions which each accept `Type` and `Dsymbol`, respectively. I don't see any `bool` in that diff.

Reply:
I thought you wanted an overload set for `create`. Ok then.

Reply:
> I'm probably missing something here.

The template just forwards to the right `__ctor`, which then does the difference between Type and Dsymbol, this saves 4 or 5 SLOCs. Also the new function is for intern use only, not even need in the cpp header.

I think it's better to keep it. Eventually I can remove the constraint.

Reply:
Ah I see what you mean now. But then it's a refactoring. Why not simply set the STC at the call site using `|=` like was done previously ?
I understand your reasoning, but I personally would be happier with a ctor overload that takes a bool. Having a special static function which uses templates to emulate an overload set is IMHO the worse approach out of the ones you mentioned.

Reply:
the more simple then is to break the guidlines and add an optional bool param to the existing ctor. :fog:

Reply:
Maybe someone else could bring a fresh point of view on this detail ?

Reply:
Granted there's only two calls to this (one for each overload), I'm not fussed keeping the setting of the storage class at the callsite, making this function redundant.

Reply:
ok done, better without new function, thanks for the fresh point of view.